src/test/java/com/google/sps/servlets/ContestServletTest.java

```diff
@@ -0,0 +1,122 @@
+package com.google.sps.servlets;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalURLFetchServiceTestConfig;
+import java.io.PrintWriter;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.*;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class ContestServletTest {
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig(), new LocalURLFetchServiceTestConfig());
+
+  @Mock HttpServletRequest httpServletRequest;
+  @Mock HttpServletResponse httpServletResponse;
+  @Mock PrintWriter printWriter;
+
+  private static Set<Long> candidateIdSet;
+  private static Entity electionEntityOne;
+  private static Entity contestEntityOne;
+  private static Entity contestEntityTwo;
+
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @BeforeClass
+  public static void classSetUp() {
+    helper.setUp();
+
+    candidateIdSet = new HashSet<Long>();
+    candidateIdSet.add(Long.valueOf(1));
+    candidateIdSet.add(Long.valueOf(2));
+    candidateIdSet.add(Long.valueOf(3));
+
+    electionEntityOne = new Entity("Election");
+    electionEntityOne.setProperty("id", "9999");
+    electionEntityOne.setProperty("name", "myElection");
+    electionEntityOne.setProperty("scope", "myScope");
+    electionEntityOne.setProperty("date", "myDate");
+    electionEntityOne.setProperty("contests", new HashSet<Long>());
+    electionEntityOne.setProperty("propositions", new HashSet<Long>());
+
+    contestEntityOne = new Entity("Contest", 1);
+    contestEntityOne.setProperty("name", "myFirstContest");
+    contestEntityOne.setProperty("candidates", candidateIdSet);
+    contestEntityOne.setProperty("description", "This contest is important.");
+
+    contestEntityTwo = new Entity("Contest", 2);
+    contestEntityTwo.setProperty("name", "mySecondContest");
+    contestEntityTwo.setProperty("candidates", new HashSet<Long>());
```

Comment:
This code should probably be in the setup method annotated @Before instead of @BeforeClass (@BeforeClass is only run once at the before running ALL the tests in this file, where as @Before is run before *each* test)

You shouldn't have to reset contests and propositions entities in @TearDown

Reply:
+1.  BeforeClass should be relatively rare usage.

Reply:
Removed method tagged @BeforeClass